### ISSUE TITLE
In-Game: Various maths functions interpret "undefined" as the number -6

### ISSUE THREAD

**Issue by KormexGit** [YoYoGames/GameMaker-Bugs#7185]
### Description
After seeing someone having a crash in GMRT due to calling abs on undefined, I was curious what doing that in GMS2 runtime did. Turns out, doing something like `round(undefined)` gives you -6. A variety of other maths functions also seem to interpret undefined as -6. I'd expect passing undefined to these functions to crash, or return undefined again, or something else that makes more sense than acting like you passed -6 lol

I got the same results for both VM and YYC.

### Steps To Reproduce
```
show_debug_message(undefined);
show_debug_message(abs(undefined));
show_debug_message(round(undefined));
show_debug_message(sign(undefined));
```
Will show 
```
undefined
6
-6
-1
```

### Which version of GameMaker are you reporting this issue for?
IDE v2024.800.0.605 Runtime v2024.800.0.628

### Which operating system(s) are you seeing the problem on?
Windows 10.0.22631.0

### Which platform(s) are you seeing the problem on?
Windows

[86e763af-c70a-4a42-a7ee-9c7d8f4eb489](https://api.gamemaker.io/api/github/downloads/86e763af-c70a-4a42-a7ee-9c7d8f4eb489)

**Comment by rwkay** [comment]
Fixed in 2024.8

**Comment by rwkay** [comment]
The error is expected...

min, max, mean and median all use arrays... I will update the fix to include that code

choose ,random, random_range, irandom, irandom_range and sqrt - they are not optimised at compile time so would not be "fixed" by this code.

**Comment by rwkay** [comment]
Fixed in 2024.8 - added handling for arrays which should fix min, max, mean and median

**Comment by scott-dunbar** [comment]
Verified fixed in  IDE v2024.800.0.616  Runtime v2024.800.0.639
